tests/e2e/e2e_test.go

```diff
@@ -17,46 +18,587 @@
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get k8s clients
+	getK8sClients()
+
+	// Create clients for apiextensions and our CRD api
+	extClient = extclient.NewForConfigOrDie(clientConfig)
+
+	// Create a namespace for the test
+	testNamespace, err = CreateTestingNS("k8s-device-plugin-e2e-test", clientSet, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get Helm client
+	helmReleaseName = "k8s-device-plugin-e2e-test" + rand.String(5)
+	getHelmClient()
+})
+
+var _ = AfterSuite(func(ctx SpecContext) {
+	By("Cleaning up namespace resources")
+	cleanupNamespaceResources(testNamespace.Name)
+
+	By("Deleting the test namespace")
+	deleteTestNamespace()
+})
+
+// Add ReportAfterSuite for logging test summary and random seed
+var _ = ReportAfterSuite("", func(report Report) {
+	// Log test summary
+	failedCount := 0
+	for _, specReport := range report.SpecReports {
+		if specReport.Failed() {
+			failedCount++
+		}
+	}
+
+	GinkgoWriter.Printf("\nTest Summary:\n")
+	GinkgoWriter.Printf("  Total Specs: %d\n", len(report.SpecReports))
+	GinkgoWriter.Printf("  Random Seed: %d\n", report.SuiteConfig.RandomSeed)
+	GinkgoWriter.Printf("  Failed: %d\n", failedCount)
+	GinkgoWriter.Printf("  Duration: %.2fs\n", report.RunTime.Seconds())
+})
+
+// getK8sClients creates the k8s clients
+func getK8sClients() {
+	var err error
+
+	// get config from kubeconfig
+	c, err := clientcmd.LoadFromFile(Kubeconfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// get client config
+	clientConfig, err = clientcmd.NewDefaultClientConfig(*c, &clientcmd.ConfigOverrides{}).ClientConfig()
+	Expect(err).NotTo(HaveOccurred())
+
+	clientSet, err = clientset.NewForConfig(clientConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Create clients for apiextensions and our CRD api
+	nfdClient = nfdclient.NewForConfigOrDie(clientConfig)
+}
+
+// getHelmClient creates a new Helm client
+func getHelmClient() {
+	var err error
+
+	// Set Helm log file
+	helmArtifactDir = filepath.Join(LogArtifactDir, "helm")
+
+	// Create a Helm client
+	err = os.MkdirAll(helmArtifactDir, 0755)
+	Expect(err).NotTo(HaveOccurred())
+
+	helmLogFile, err = os.OpenFile(filepath.Join(LogArtifactDir, "helm_logs"), os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	Expect(err).NotTo(HaveOccurred())
+
+	helmLogger = log.New(helmLogFile, fmt.Sprintf("%s\t", testNamespace.Name), log.Ldate|log.Ltime)
+
+	helmRestConf := &helm.RestConfClientOptions{
+		Options: &helm.Options{
+			Namespace:        testNamespace.Name,
+			RepositoryCache:  "/tmp/.helmcache",
+			RepositoryConfig: "/tmp/.helmrepo",
+			Debug:            true,
+			DebugLog:         helmLogger.Printf,
+		},
+		RestConfig: clientConfig,
+	}
+
+	helmClient, err = helm.NewClientFromRestConf(helmRestConf)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// getTestEnv gets the test environment variables
+func getTestEnv() {
+	defer GinkgoRecover()
+	var err error
+
+	Kubeconfig = os.Getenv("KUBECONFIG")
+	Expect(Kubeconfig).NotTo(BeEmpty(), "KUBECONFIG must be set")
```

Comment:
Nit: In `nvidia-container-toolkit` we have the concept of a REQUIRED envvar that we encapsulated in a function.